examples/hello/server.c

```diff
@@ -0,0 +1,152 @@
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "hello.h"
+
+/*
+ * this file represent a server receving a message from a client
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_pt_index_t pti;
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_le_t leh;
+	long num_messages;
+	char *p;
+	char msg[PTL_EV_STR_SIZE];
+	char data[50];
+	/* represent the memory used to receive the message */
+	ptl_le_t le = { .start = data,
+			.length = sizeof(data),
+			.ct_handle = PTL_CT_NONE,
+			.uid = PTL_UID_ANY,
+			.options = PTL_LE_OP_PUT };
+
+	if (argc < 2) {
+		num_messages = 0;
+		printf("Usage: client <number of possible received messages>\n default : 0\n");
+	} else {
+		num_messages = strtol(argv[1], &p, 10);
+		if (*p != '\0') {
+			fprintf(stderr,
+				"An invalid character was found before the end of the string");
+		}
+	}
+
+	printf(" \n--- server interface --- \n \n");
+
+	ret = PtlInit();
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		return 1;
+	}
+
+	ret = PtlNIInit(PTL_IFACE_DEFAULT, PTL_NI_NO_MATCHING | PTL_NI_PHYSICAL, SERVER_PID, NULL,
+			NULL, &nih);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlNIInit failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto fini;
+	}
+
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	ret = PtlPTAlloc(nih, 0, eqh, 0, &pti);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlPTAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto eq_free;
+	}
+
+	/* prepare an events buffer to receive a message represented by le */
+	ret = PtlLEAppend(nih, pti, &le, PTL_PRIORITY_LIST, NULL, &leh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+
+	/* get server pid and server nid so the client know the server id */
+	ret = PtlGetId(nih, &id);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlGetId failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto le_unlink;
+	}
+	printf("Server Pid : %d \n", id.phys.pid);
+	printf("Server Nid : %d", id.phys.nid);
+
+	while (num_messages >= 0) {
+		/* wait until a message arrives */
+		printf("\n");
+		ret = PtlEQWait(eqh, &ev);
+		if (ret != PTL_OK) {
+			fprintf(stderr, "PtlEQWait failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+			res = 1;
+			goto le_unlink;
+		}
+		PtlEvToStr(0, &ev, msg);
+		printf("Event : %s", msg);
+		if (ev.type == PTL_EVENT_PUT) {
+			if (ev.rlength + 1 >= sizeof(data)) {
+				fprintf(stderr,
+					"The received message is too long to be entirely readed");

```

Comment:
in that case, the string null terminating character is not set, so the "message" printf below will read beyond data buffer